inst/examples/multiTrackApp/server.R

```diff
@@ -0,0 +1,335 @@
+function(input, output, session) {
+  all_reactive_values <- reactiveValues(
+    single_composed_track = NULL,
+    single_composed_views = NULL,
+    single_track = NULL,
+    composed_track = NULL,
+    composed_views = NULL,
+    track1 = NULL,
+    track2 = NULL,
+    track3 = NULL,
+    track4 = NULL
+  )
+
+  observeEvent(input$reset_rule_mark, {
+    zoom_to_extent(
+      component_id = "component_1",
+      view_id = "track1",
+      duration = 2000
+    )
+  })
+
+  observeEvent(input$layout_single, {
+
+    all_reactive_values$single_track <- add_single_track(
+      id = "track1",
+      data = track_data(
+        url = "https://server.gosling-lang.org/api/v1/tileset_info/?d=cistrome-multivec",
+        type = "multivec",
+        row = "sample",
+        column = "position",
+        value = "peak",
+        categories = c("sample 1", "sample 2", "sample 3", "sample 4"),
+        binSize = 4,
+      ),
+      mark = "rect",
+      x = visual_channel_x(field = "start", type = "genomic", axis = "top"),
+      xe = visual_channel_x(field = "end", type = "genomic"),
+      row = visual_channel_row(
+        field = "sample",
+        type = "nominal",
+        legend = TRUE
+      ),
+      color = visual_channel_color(
+        field = "peak",
+        type = "quantitative",
+        legend = TRUE
+      ),
+      tooltip = visual_channel_tooltips(
+        visual_channel_tooltip(field = "start", type = "genomic", alt = "Start Position"),
+        visual_channel_tooltip(field = "end", type = "genomic", alt = "End Position"),
+        visual_channel_tooltip(
+          field = "peak",
+          type = "quantitative",
+          alt = "Value",
+          format = "0.2"
+        )
+      ),
+      width = 600,
+      height = 130
+    )
+
+    all_reactive_values$single_composed_track <- compose_view(
```

Comment:
Why we want to store this in a `reactiveValues` object instead of using a variable here?

Reply:
Yep, good catch. :)